common/src/main/java/io/seata/common/loader/EnhancedServiceLoader.java

```diff
@@ -336,11 +422,51 @@ private static void loadFile(Class<?> service, String dir, ClassLoader classLoad
         }
     }
 
+    private ExtensionURL getExtensionURL(Class<?> clazz) {
+        String serviceName = null;
+        String typeName = null;
+        Integer priority = 0;
+        Scope scope = Scope.SINGLETON;
+        LoadLevel loadLevel = clazz.getAnnotation(LoadLevel.class);
+        if (loadLevel == null) {
+            typeName = clazz.getTypeName();
+        } else {
+            serviceName = loadLevel.name();
+            priority = loadLevel.order();
+            scope = loadLevel.scope();
+        }
+        ExtensionURL result = new ExtensionURL(serviceName, typeName, priority, scope);
+        //do cache
+        extensionClassUrlMap.put(clazz, result);
+        if (loadLevel != null) {
+            if (extensionNameUrlsMap.containsKey(serviceName)) {
+                List<ExtensionURL> urls = extensionNameUrlsMap.get(serviceName);
+                urls.add(result);
+            } else {
+                List<ExtensionURL> urls = new ArrayList<>();
+                urls.add(result);
+                extensionNameUrlsMap.put(serviceName, urls);
+            }
+        }
+        if (priority >= highestLoadPriority) {
+            highestLoadPriority = priority;
+            defaultExtensionURL = result;
+            return defaultExtensionURL;
```

Comment:
remove

Reply:
The highest priority based on the order param in LoadLevel matches the current class's  defaultExtensionURL,we cached them together so that next time this extension is loaded,we can directly find the default extension by cache.